Cargo.toml

```diff
@@ -99,3 +99,15 @@ multistream-select = { version = "0.13.0", path = "misc/multistream-select" }
 quick-protobuf-codec = { version = "0.2.0", path = "misc/quick-protobuf-codec" }
 quickcheck = { package = "quickcheck-ext", path = "misc/quickcheck-ext" }
 rw-stream-sink = { version = "0.4.0", path = "misc/rw-stream-sink" }
+multiaddr = { git = "https://github.com/thomaseizinger/rust-multiaddr", branch = "typesafe-p2p" }
+multihash = "0.19.0"
+
+
+[patch.crates-io]
+
+# Patch away `libp2p-idnentity` in our dependency tree with the workspace version.
+# `libp2p-identity` is a leaf dependency and used within `rust-multiaddr` which is **not** part of the workspace.
+# As a result, we cannot just reference the workspace version in our crates because the types would mismatch with what
+# we import via `rust-multiaddr`.
+# This is expected to stay here until we move `libp2p-identity` to a separate repository which makes the dependency relationship more obvious.
+libp2p-identity = { path = "identity" }
```

Comment:
@mxinden I did not expect that we need this but it actually makes sense. We can't just depend on `libp2p-idenity` via `path` in here because it will mismatch with what is coming from `multiaddr`.

Reply:
This makes sense. I think `cargo publish` will ignore all `patch` sections, thus this does not affect our published crates.